### PR TITLE
docs(option): update WithDefaultEndpointTemplate docs

### PR DESCRIPTION
**Pull request by quartzmo**
Follow-up to #2335.

### CODE REVIEW

File: option/internaloption/internaloption.go [hunk 1]
```diff
@@ -163,6 +166,10 @@ func (w withDefaultUniverseDomain) Apply(o *internal.DialSettings) {
 
 // EnableJwtWithScope returns a ClientOption that specifies if scope can be used
 // with self-signed JWT.
+//
+// EnableJwtWithScope is ignored when option.WithUniverseDomain is set
+// to a non-GDU universe domain. For non-GDU universe domains, token exchange is
```
Comment (codyoss) [inline option/internaloption/internaloption.go:171]: What is a GDU, would a user know this term?

Comment (quartzmo) [reply option/internaloption/internaloption.go:171]: Good call, I'll expand it.

Comment (quartzmo) [reply option/internaloption/internaloption.go:171]: Actually, this is internaloption, is it ok here? Or do you think I should spell it out for maintainers as well.

Comment (codyoss) [reply option/internaloption/internaloption.go:171]: I think it is still worth while as I am not sure what it stand for

Comment (quartzmo) [reply option/internaloption/internaloption.go:171]: "Google Default Universe" fwiw

**Review by codyoss — APPROVED** [review]

File: option/internaloption/internaloption.go (hunk 1 above)
Comment (quartzmo) [reply option/internaloption/internaloption.go]: Fixed, thanks!

**Pull request merged**